gsy_framework/sim_results/kpi.py

```diff
@@ -368,7 +384,19 @@ def restore_area_results_state(self, area_dict: Dict, last_known_state_data: Dic
                 last_known_state_data["total_self_consumption_wh"]
             self.state[area_dict["uuid"]].self_consumption_buffer_wh = \
                 last_known_state_data["self_consumption_buffer_wh"]
-
+        if area_dict["uuid"] not in self.savings_state:
+            if not has_grand_children(area_dict):
```

Comment:
```suggestion
        if not (area_dict["uuid"] in self.savings_state or has_grand_children(area_dict)):
```

Reply:
Update: Nit, can be skipped

Reply:
Will merge it now, and drop a comment to address in follow up ticket to refactor and add unittest